data/pigui/modules/station-view/04-shipMarket.lua

```diff
@@ -159,15 +162,135 @@ local function buyShip (mkt, sos)
 	mkt.popup:open()
 end
 
-local yes_no = function (binary)
+local FormatAndCompareShips = utils.class("ui.ShipMarket.FormatAndCompareShips")
+
+function FormatAndCompareShips:beginTable()
+	if not ui.beginTable("specs", 7 ) then
+		return false
+	end
+
+	ui.tableSetupColumn("name1")
+	ui.tableSetupColumn("body1")
+	ui.tableSetupColumn("indicator1", {"WidthFixed"})
+	ui.tableSetupColumn("gap", {"WidthFixed"})
+	ui.tableSetupColumn("name2")
+	ui.tableSetupColumn("body2")
+	ui.tableSetupColumn("indicator2", {"WidthFixed"})
+
+	return true
+end
+
+function FormatAndCompareShips:compare_and_draw_column(desc, str, a, b)
+	compare = a-b
+
+	if self.column == 0 then
+		ui.tableNextRow()
+	end
+
+	ui.tableSetColumnIndex(0 + self.column)
+	ui.textColored(styleColors.gray_300, desc)
+
+	ui.tableSetColumnIndex(1 + self.column)
+
+	if compare < 0 then
+		ui.textAlignedColored(str, 1.0, ui.theme.colors.shipmarketCompareWorse)
+		ui.tableSetColumnIndex(2 + self.column)
+		ui.icon( ui.theme.icons.shipmarket_compare_worse, Vector2(ui.getTextLineHeight()), ui.theme.colors.shipmarketCompareWorse)		
+		
+	elseif compare > 0 then
+		ui.textAlignedColored(str, 1.0,  ui.theme.colors.shipmarketCompareBetter)
+		ui.tableSetColumnIndex(2 + self.column)
+		ui.icon( ui.theme.icons.shipmarket_compare_better, Vector2(ui.getTextLineHeight()), ui.theme.colors.shipmarketCompareBetter)			
+	else
+		ui.textAligned(str, 1.0)
+		ui.tableSetColumnIndex(2 + self.column)
+		ui.dummy( Vector2(ui.getTextLineHeight()) )		
+	end
+
+	if self.column == 0 then
+		self.column = 4
+	else
+		self.column = 0
+	end
+end
+
+function FormatAndCompareShips:draw_hyperdrive_cell(desc)
+	local hyperdrive_str = self.def.hyperdriveClass > 0 and
+		Equipment.hyperspace["hyperdrive_" .. self.def.hyperdriveClass]:GetName() or l.NONE
+
+	self:compare_and_draw_column( desc, hyperdrive_str, self.def.hyperdriveClass, self.b.def.hyperdriveClass )
+end
+
+function FormatAndCompareShips:get_value(key)
+	v = self[key]
+	if nil == v then
+		v = self.def[key]
+	end
+	return v
+end
+
+function FormatAndCompareShips:draw_tonnage_cell(desc, key)
+	self:compare_and_draw_column( desc, Format.MassTonnes(self:get_value(key)), self:get_value(key), self.b:get_value(key) )
+end
+
+function FormatAndCompareShips:draw_accel_cell(desc, thrustKey, massKey, multiplier)
+	local accelA = multiplier * self.def.linearThrust[thrustKey] / (-9.81*1000*(self:get_value(massKey)))
+	local accelB = multiplier * self.b.def.linearThrust[thrustKey] / (-9.81*1000*(self.b:get_value(massKey)))
+	self:compare_and_draw_column( desc, Format.AccelG(accelA), accelA, accelB )
+end
+
+function FormatAndCompareShips:draw_deltav_cell(desc, massNumeratorKey, massDenominatorKey)
+	local deltavA = self.def.effectiveExhaustVelocity * math.log( self:get_value(massNumeratorKey), self.b:get_value(massDenominatorKey) )
+	local deltavB = self.b.def.effectiveExhaustVelocity * math.log( self.b:get_value(massNumeratorKey), self.b:get_value(massDenominatorKey) )	
```

Comment:
This deltaV calculation is also incorrect. It should be `math.log( massNumerator / massDenominator )`, with the empty second parameter defaulting to *e*. Lua's log function takes the base of the logarithm in the second parameter, not a divisor.

Reply:
All done and completed.  Tooltips are a great addition :-)